consensus/polybft/consensus_runtime.go

```diff
@@ -156,15 +163,16 @@ func (c *consensusRuntime) getEpoch() *epochMetadata {
 	return c.epoch
 }
 
-// getLastBuiltBlockAndEpoch returns last build block and current epochMetadata in a thread-safe manner.
-func (c *consensusRuntime) getLastBuiltBlockAndEpoch() (*types.Header, *epochMetadata) {
+// getSyncData returns last build block, proposer snapshot and current epochMetadata in a thread-safe manner.
+func (c *consensusRuntime) getSyncData() (*types.Header, *epochMetadata, *ProposerSnapshot) {
 	c.lock.RLock()
 	defer c.lock.RUnlock()
 
 	lastBuiltBlock, epoch := new(types.Header), new(epochMetadata)
 	*lastBuiltBlock, *epoch = *c.lastBuiltBlock, *c.epoch
```

Comment:
it's not safe copy. Could you use c.lastBuiltBlock.Copy?

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1047/commits/70d5daed07acc17e48bd989c49a2921c8dc76d2a